lib/global_uid/allocations.rb

```diff
@@ -0,0 +1,28 @@
+class Allocations
+  attr_reader :allocations, :max_size, :incrementing_by
+
+  def initialize(max_size: 5, incrementing_by:)
+    @allocations = []
+    @max_size = max_size
+    @incrementing_by = incrementing_by
+  end
+
+  def add(identifier)
+    allocations.shift if allocations.size >= max_size
+    allocations << identifier
+    valid_increment?
+  end
+  alias_method :<<, :add
```

Comment:
Maybe `add` should raise on failure instead of returning `false` so it doesn't get accidentally ignored. Or there could be an `add!`, and `<<` could be an alias for `add!`.

Having `<<` return `true/false` seems a little odd since Ruby's `Array <<` method returns the array itself.

Reply:
👍 thanks